state/factory/factory.go

```diff
@@ -652,6 +653,11 @@ func readState(tlt trie.TwoLayerTrie, ns string, key []byte, s interface{}) erro
 	return state.Deserialize(s, data)
 }
 
+func toLayerTwoKey(input []byte) []byte {
```

Comment:
add if( len(input) != 20 )?
since most keys now (account, contract) are 20-bytes long

Reply:
also maybe name it toLegacyKey()? we use Legacy to mean 20-bytes key, like:

func LegacyKeyOption(key hash.Hash160) StateOption